prometheus-metrics-exporter-httpserver/src/main/java/io/prometheus/metrics/exporter/httpserver/HTTPServer.java

```diff
@@ -55,7 +55,13 @@ private HTTPServer(PrometheusProperties config, ExecutorService executorService,
         registerHandler("/", defaultHandler == null ? new DefaultHandler() : defaultHandler, authenticator);
         registerHandler("/metrics", new MetricsHandler(config, registry), authenticator);
         registerHandler("/-/healthy", new HealthyHandler(), authenticator);
-        this.server.start();
+        try {
+            executorService.submit(() -> this.server.start()).get();
```

Comment:
Remove the `get` call if the `Future` is not being used.

Reply:
Hey @dhoard,
I'd like to push back on this. Starting the HTTP server might fail and throw an exception, e. g. when the port is already in use. Unless I call `.get()` on this `Future`, this error would be silently discarded, which is something that should be avoided.

Reply:
@mberndt123 good catch/makes sense. Can you add a comment to the code? ... anyone looking at the code wouldn't get this nuance - I didn't.

Reply:
OK